editHack.js

```diff
@@ -0,0 +1,49 @@
+const mongoose = require("mongoose");
+
+let conn = null;
+const url = `mongodb://${process.env.MONGO_USER}:${process.env.MONGO_PASSWORD}@ds157136.mlab.com:57136/hackone`;
+
+export const edit = async (event, context) => {
+  context.callbackWaitsForEmptyEventLoop = false;
+  const data = JSON.parse(event.body);
+  const { goal, title, description, hackId } = data;
+  let update = {};
+  if (conn == null) {
+    conn = await mongoose.createConnection(url, {
+      bufferCommands: false, // Disable mongoose buffering
+      bufferMaxEntries: 0 // and MongoDB driver buffering
+    });
+    conn.model(
+      "Hack",
+      new mongoose.Schema({ title: String, description: String, goal: String })
+    );
+  }
+  const Hack = conn.model("Hack");
+
+  // guard against empty fields
+  if (goal) update.goal = goal;
```

Comment:
so `undefined`, `null` or `""` will all be missing from the document? these are not guarded against in the addHack lambda (https://github.com/hackthehack/hack-zero-v2-backend/blob/master/addHack.js)

Reply:
Valid point. Might include a default value when document is created in mongodb. Will address that in addHack.js before merge to master👍